'use strict';
const {{pascalName}}Model = use("App/Models/{{modelFolder}}/{{pascalName}}");
const CustomException = use('CustomException')
const ControllerHelper = use('ControllerHelper')

class Show {

  constructor(transaction, user) {
    this.transaction = transaction;
    this.user = user;
  }

  async execute(form) {
    const query = {{pascalName}}Model
      .query(this.transaction)
      .where('{{primaryColumn}}', form.{{primaryColumn}})

       let response = await ControllerHelper
            .populate(query, form.populates)
            .first();

    if (!response) {
      throw new CustomException("{{pascalName}} not found", 400, "{{uppercase}}_NOT_FOUND")
    }
    return response;
  }
}

module.exports = Show;
